### ISSUE TITLE
Move check_random_state in LeaveOneOutEncoder

### ISSUE THREAD

**Issue by janmotl** [scikit-learn-contrib/categorical-encoding#103]
Following code in LeaveOneOutEncoder:
```
self.random_state_ = check_random_state(self.random_state)
```
should possibly be moved from `fit()` method into `transform()` method, where the random number generator is actually used. Reasoning: I would  expect the following code to pass:
```
enc = encoders.LeaveOneOutEncoder(random_state=2001, randomized=True)
enc.fit(X, y)
a = enc.transform(X_t, y_t)
b = enc.transform(X_t, y_t)
self.assertTrue(a.equals(b))
```
but currently it does not pass the test.

A possible place where to move it:
```
if self.randomized and y is not None:
    random_state_generator = check_random_state(self.random_state)
    X[switch.get('col')] = (X[switch.get('col')] * random_state_generator.normal(1., self.sigma, X[switch.get('col')].shape[0]))
```

**Comment by wdm0006** [comment]
Agreed

**Comment by janmotl** [comment]
Solved with #110
